binary_search_networks/pipeline.py

```diff
@@ -41,4 +43,4 @@ def run_model(n):
     plt.savefig('results/accuracy', dpi=300)
```

Comment:
plt.savefig is not correct here, you have the pass the .png or .jpg extension. Also, please add the model name to the file name so that we know which one is which.